clang/include/clang/CIR/Dialect/IR/CIRTypes.h

```diff
@@ -20,10 +20,122 @@
 #include "clang/CIR/Interfaces/ASTAttrInterfaces.h"
 
 //===----------------------------------------------------------------------===//
-// CIR Dialect Types
+// CIR Dialect Tablegen'd Types
 //===----------------------------------------------------------------------===//
 
 #define GET_TYPEDEF_CLASSES
 #include "clang/CIR/Dialect/IR/CIROpsTypes.h.inc"
 
+//===----------------------------------------------------------------------===//
+// CIR StructType
+//
+// The base type for all RecordDecls.
+//===----------------------------------------------------------------------===//
+
+namespace mlir {
+namespace cir {
+
+namespace detail {
+struct StructTypeStorage;
+} // namespace detail
+
+class StructType
```

Comment:
Could you add a FIXME so that this can be made back into tablegen when support for mutable shows up?